RecoMET/METPUSubtraction/python/MVAMETConfiguration_cff.py

```diff
@@ -0,0 +1,248 @@
+
+    # get jets for T1 Correction
+    from RecoJets.JetProducers.ak4PFJets_cfi import ak4PFJets
+    from JetMETCorrections.Configuration.JetCorrectors_cff import ak4PFCHSL1FastL2L3Corrector,ak4PFCHSL3AbsoluteCorrector,ak4PFCHSL2RelativeCorrector,ak4PFCHSL1FastjetCorrector, ak4PFCHSL1FastL2L3ResidualCorrector, ak4PFCHSResidualCorrector
+    process.ak4PFCHSL1FastL2L3Corrector = ak4PFCHSL1FastL2L3Corrector
+    process.ak4PFCHSL1FastL2L3ResidualCorrector = ak4PFCHSL1FastL2L3ResidualCorrector
+    process.ak4PFCHSResidualCorrector = ak4PFCHSResidualCorrector
+    process.ak4PFCHSL3AbsoluteCorrector = ak4PFCHSL3AbsoluteCorrector
+    process.ak4PFCHSL2RelativeCorrector = ak4PFCHSL2RelativeCorrector
+    process.ak4PFCHSL1FastjetCorrector = ak4PFCHSL1FastjetCorrector 
+
+
+    for met in ["pfMET", "pfTrackMET", "pfNoPUMET", "pfPUCorrectedMET", "pfPUMET", "pfChargedPUMET", "pfNeutralPUMET", "pfNeutralPVMET", "pfNeutralUnclusteredMET"]:
+        # create PF METs
+        setattr(process, met, pfMet.clone(src = cms.InputTag(met+"Cands"), alias = cms.string(met)))
+        # create Jets
+        setattr(process, "ak4JetsFor"+met, ak4PFJets.clone(src = cms.InputTag(met+"Cands")))
+        setattr(process, "corr"+met, corrPfMetType1.clone(src = cms.InputTag("ak4JetsFor"+met)))
+        # derive corrections and apply them
+        setattr(process, met+"T1", pfMetT1.clone( src= cms.InputTag(met), srcCorrections=cms.VInputTag(cms.InputTag("corr"+met, "type1"))))
+        # convert METs to pat objects
+        setattr(process, "pat"+met,      patMETsForMVA.clone(metSource = cms.InputTag(met)))
+        setattr(process, "pat"+met+"T1", patMETsForMVA.clone(metSource = cms.InputTag(met+"T1")))
+
+    process.pfChs = cms.EDProducer("CandViewMerger", src = cms.VInputTag(          
+                                                                                          cms.InputTag("pfChargedPV"),
+                                                                                          cms.InputTag("neutralInJets", "neutralPassingPUIDJets"),
+                                                                                          cms.InputTag("neutralInJets", "neutralFailingPUIDJets"),
+                                                                                          cms.InputTag("neutralInJets", "neutralParticlesUnclustered")
+    ))
+    process.ak4JetsForpfMET.src = cms.InputTag("pfChs")
+
+    ### MVA MET
+    process.MVAMET = cms.EDProducer("MVAMET",                                                
+                                    debug = cms.bool(debug),
+                                    requireOS = cms.bool(True),
+                                    combineNLeptons = cms.int32(2),
+                                    MVAMETLabel = cms.string("MVAMET"),
+                                    srcMETs      = cms.VInputTag(
+                                                                 cms.InputTag("slimmedMETs"),
```

Comment:
Here it's specified, which collections should be available in the MVA MET. Which ones are actually used has to be specified during the training. This information is saved in the "weight file" and then automatically picked up correctly by the MVAMET plugin